pkg/api/testing/.import-restrictions

```diff
@@ -1,12 +1,5 @@
-{
-	"Rules": [
-	  {
-		"SelectorRegexp": "k8s[.]io/kubernetes/cmd",
-		"AllowedPrefixes": [
-		    "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
-		],
-		"ForbiddenPrefixes": [
-		]
-	  }
-	]
-}
\ No newline at end of file
+rules:
+  # override pkg/ import restriction on cmd/ for pkg/api/testing
+  - selectorRegexp: k8s[.]io/kubernetes/cmd
+    allowedPrefixes:
+      - k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm
```

Comment:
@neolit123 I didn't change this one at all, but wasn't sure whether it indicated any tangles for kubeadm, seems like we wouldn't want k8s.io/kubernetes to depend on kubeadm?

Reply:
yes, i don't how `pkg/api/testing` depends on `k8s.io/kubernetes/cmd/kubeadm`. does import boss complain without this rule? i see a commit by dims that added that https://github.com/kubernetes/kubernetes/commit/21fdd1c5ed85fe18cd1dacbdb2e402fbb92239e6#diff-4110867282788a4ac1cc35e7fba1463a

the gendocs and genman packages do import `cmd/kubeadm`, but that is known:
```
grep "cmd/kubeadm" * -rnI | grep -v "^cmd/kubeadm"
build/BUILD:134:        "//cmd/kubeadm",
build/visible_to/BUILD:59:        "//cmd/kubeadm/app/cmd",
build/dependencies.yaml:19:    - path: cmd/kubeadm/app/constants/constants.go
build/dependencies.yaml:68:    - path: cmd/kubeadm/app/constants/constants.go
CHANGELOG/CHANGELOG-1.13.md:1128:  - support for the `v1alpha3` configuration file format is deprecated and will be removed in 1.14. Use `kubeadm config migrate` to migrate `v1alpha3` configuration files to `v1beta1`, which provides improvements in image repository management, addons configuration, and other areas. The documentation for `v1beta1` can be found here: https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1
CHANGELOG/CHANGELOG-1.13.md:1183:- kubeadm’s configuration API is now `v1beta1`. The new configuration format provides improvements in - image repository management, addons configuration, and other areas. We encourage `v1alpha3` users to migrate to this configuration API using `kubeadm config migrate`, as `v1alpha3` will be removed in 1.14. The documentation for `v1beta1` can be found here: https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1
CHANGELOG/CHANGELOG-1.15.md:2123:* In case kubeadm can't access the current Kubernetes version remotely and fails to parse the git-based version it falls back to a static predefined value of k8s.io/kubernetes/cmd/kubeadm/app/constants.CurrentKubernetesVersion. ([#72454](https://github.com/kubernetes/kubernetes/pull/72454), [@rojkov](https://github.com/rojkov))
cmd/BUILD:28:        "//cmd/kubeadm:all-srcs",
cmd/genkubedocs/BUILD:31:        "//cmd/kubeadm/app/cmd:go_default_library",
cmd/genkubedocs/gen_kube_docs.go:30:	kubeadmapp "k8s.io/kubernetes/cmd/kubeadm/app/cmd"
cmd/genman/BUILD:24:        "//cmd/kubeadm/app/cmd:go_default_library",
cmd/genman/gen_kube_man.go:34:	kubeadmapp "k8s.io/kubernetes/cmd/kubeadm/app/cmd"
hack/lib/golang.sh:77:    cmd/kubeadm
hack/lib/golang.sh:125:    cmd/kubeadm
hack/.golint_failures:4:cmd/kubeadm/app/apis/kubeadm/v1beta1
hack/.golint_failures:5:cmd/kubeadm/app/apis/kubeadm/v1beta2
hack/make-rules/test.sh:43:          -o -path './cmd/kubeadm/test/*' \
hack/make-rules/test-cmd.sh:127:  make all -C "${KUBE_ROOT}" WHAT=cmd/kubeadm
hack/make-rules/test-cmd.sh:132:    WHAT=k8s.io/kubernetes/cmd/kubeadm/test/cmd
make-kubeadm.sh:6:make all WHAT=cmd/kubeadm
pkg/api/testing/.import-restrictions:6:		    "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"

```

Reply:
ok, i don't think this import restriction is needed as is.
sent PR with a related cleanup here:
https://github.com/kubernetes/kubernetes/pull/90385